### ISSUE TITLE
 QueryFailedError: relation "migrations" already exists

### ISSUE THREAD

**Issue by webmobiles** [typeorm/typeorm#3107]
**Issue type:**

[x] bug report

**Database system/driver:**

[x] `postgres`

**TypeORM version:**

[ ] `latest`

**Steps to reproduce or a small repository showing the problem:**

I've just delete my public schema in postgres and rename to 'schematest'

on create connection:

```

      await createConnection({
        type: 'postgres',
        schema: 'hbd',
        url: process.env.DATABASE_URL,
```
my DATABASE_URL=postgresql://myuser:testtest@localhost:5432/bpi

but when i run the app i get:
```

15:35:11 - error: An error occured during connection to the database: QueryFailedError: relation "migrations" already exists
    at new QueryFailedError (//src/error/QueryFailedError.ts:7:9)
    at Query.callback (//src/driver/postgres/PostgresQueryRunner.ts:175:30)
    at Query.handleError (//node_modules/pg/lib/query.js:143:17)
    at Connection.connectedErrorHandler (/node_modules/pg/lib/client.js:132:26)
    at emitOne (events.js:116:13)
    at Connection.emit (events.js:211:7)
    at Socket.<anonymous> (//node_modules/pg/lib/connection.js:123:12)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at addChunk (_stream_readable.js:263:12)
    at readableAddChunk (_stream_readable.js:250:11)
    at Socket.Readable.push (_stream_readable.js:208:10)
    at TCP.onread (net.js:607:20)

```
any idea what is happening ? 
or where on the code is trying to create 'migrations' table again?

**Comment by Kononnable** [comment]
@webmobiles 
Did you resolve your issue? If not we would probably need some more details - like if you have schema sync or similar options enabled.

**Comment by webmobiles** [comment]
I've reinstalled la db and was working ok.

**Comment by jtushman** [comment]
In case someone else finds this -- I ran into this issue when I had multiple users for the db -- and the one running the migration didn't have permissions to alter tables etc ...

**Comment by amiiit** [comment]
I hit this too, and debugging TypeORMs code I realized the core of the issue here is that
```
connection.createQueryRunner().hasTable('migrations')
```
returned `false`.

After reading here, I assumed that the reason for that is table ownership. I have no idea though how this happened all of a sudden. I did do a big refactor of my application but I didn't consciously do anything related to ownership of things in the DB.

To fix, I logged into the DB as a super user and

```
$ grant ALL on ALL tables in schema public to app;
```
`app` being the user name that your TypeORM application is connecting with.

Of course you got to be aware of security issues that might be relevant for you, giving your app permissions to do just about everything... I'm happy to get feedback on how to do this better though.

@Kononnable FYI

This seems to fix the issue. So the real issue here is TypeORM not checking whether the user has the right permissions, or just shooting an error messages that mentions this possible cause of the error.
